src/cli/bunx_command.zig

```diff
@@ -209,11 +209,20 @@ pub const BunxCommand = struct {
             update_request.name = "typescript";
         }
 
-        const initial_bin_name = if (strings.eqlComptime(update_request.name, "typescript"))
+        var initial_bin_name = if (strings.eqlComptime(update_request.name, "typescript"))
             "tsc"
         else
             update_request.name;
 
+        if (strings.startsWithChar(update_request.name, '@')) {
+            // Should be unreachable
+            if (!strings.containsChar(update_request.name, '/')) {
+                Output.prettyErrorln("<r><red>error<r><d>:<r> Invalid package name: {s}", .{update_request.name});
+                Global.exit(1);
+            }
+
+            initial_bin_name = update_request.name[strings.indexOfChar(update_request.name, '/').? + 1 ..];
+        }
```

Comment:
The following patch (on the original `bunx_command.zig`) should take care of both issues you have identified:
```diff
--- a/src/cli/bunx_command.zig
+++ b/src/cli/bunx_command.zig
@@ -204,6 +204,8 @@ pub const BunxCommand = struct {
 
         const initial_bin_name = if (strings.eqlComptime(update_request.name, "typescript"))
             "tsc"
+        else if (strings.lastIndexOfChar(update_request.name, '/')) |index|
+            update_request.name[index + 1 ..]
         else
             update_request.name;
 
@@ -365,7 +367,7 @@ pub const BunxCommand = struct {
             },
         }
 
-        absolute_in_cache_dir = std.fmt.bufPrint(&absolute_in_cache_dir_buf, "{s}/node_modules/.bin/{s}", .{ bunx_cache_dir, update_request.name }) catch unreachable;
+        absolute_in_cache_dir = std.fmt.bufPrint(&absolute_in_cache_dir_buf, "{s}/node_modules/.bin/{s}", .{ bunx_cache_dir, initial_bin_name }) catch unreachable;
 
         // Similar to "npx":
         //
```

Reply:
Actually, in the meantime I got it working by doing the split logic in `getBinNameFromSubpath`, but your code is shorter/easier to understand, so I'll use that, tnx a lot 🤗